pkg/controller/operators/olm/operatorgroup.go

```diff
@@ -422,86 +444,110 @@ func (a *Operator) ensureTenantRBAC(operatorNamespace, targetNamespace string, c
+			fetchedCSV.Annotations = newCSV.Annotations
+			// CRs don't support strategic merge patching, but in the future if they do this should be updated to patch
+			logger.Debug("updating target CSV")
+			if _, err := a.client.OperatorsV1alpha1().ClusterServiceVersions(namespace).Update(fetchedCSV); err != nil {
+				logger.WithError(err).Error("update target CSV failed")
 				return err
 			}
-			createdCSV.Status.Reason = v1alpha1.CSVReasonCopied
-			createdCSV.Status.Message = fmt.Sprintf("The operator is running in %s but is managing this namespace", csv.GetNamespace())
-			createdCSV.Status.LastUpdateTime = timeNow()
-			if _, err := a.client.OperatorsV1alpha1().ClusterServiceVersions(ns).UpdateStatus(createdCSV); err != nil {
-				a.Log.Errorf("Status update for CSV failed: %v", err)
+		}
+
+		logger.Debug("checking status")
+		newCSV.Status = csv.Status
+		newCSV.Status.Reason = v1alpha1.CSVReasonCopied
+		newCSV.Status.Message = fmt.Sprintf("The operator is running in %s but is managing this namespace", csv.GetNamespace())
+
+		if !reflect.DeepEqual(fetchedCSV.Status, newCSV.Status) {
+			logger.Debug("updating status")
+			// Must use fetchedCSV because UpdateStatus(...) checks resource UID.
+			fetchedCSV.Status = newCSV.Status
+			fetchedCSV.Status.LastUpdateTime = timeNow()
+			if _, err := a.client.OperatorsV1alpha1().ClusterServiceVersions(namespace).UpdateStatus(fetchedCSV); err != nil {
+				logger.WithError(err).Error("status update for target CSV failed")
 				return err
 			}
+		}
+
+	} else if k8serrors.IsNotFound(err) {
+		newCSV.SetNamespace(namespace)
+		newCSV.SetResourceVersion("")
 
-		} else if err != nil {
-			logger.WithError(err).Error("couldn't get CSV")
+		logger.Debug("copying CSV")
+		createdCSV, err := a.client.OperatorsV1alpha1().ClusterServiceVersions(namespace).Create(newCSV)
+		if err != nil {
+			a.Log.Errorf("Create for new CSV failed: %v", err)
+			return err
+		}
+		createdCSV.Status.Reason = v1alpha1.CSVReasonCopied
+		createdCSV.Status.Message = fmt.Sprintf("The operator is running in %s but is managing this namespace", csv.GetNamespace())
+		createdCSV.Status.LastUpdateTime = timeNow()
+		if _, err := a.client.OperatorsV1alpha1().ClusterServiceVersions(namespace).UpdateStatus(createdCSV); err != nil {
+			a.Log.Errorf("Status update for CSV failed: %v", err)
 			return err
 		}
+
+	} else if err != nil {
+		logger.WithError(err).Error("couldn't get CSV")
+		return err
 	}
 	return nil
 }
 
+// TODO: do we want to do this? or just let the dangling CSV clean it up
+func (a *Operator) pruneFromNamespace(operatorGroupName, namespace string) error {
+	fetchedCSVs, err := a.lister.OperatorsV1alpha1().ClusterServiceVersionLister().ClusterServiceVersions(namespace).List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	errlist := []error{}
+	for _, csv := range fetchedCSVs {
+		if csv.IsCopied() && csv.GetAnnotations()[v1alpha2.OperatorGroupAnnotationKey] == operatorGroupName {
```

Comment:
I think for the sake of completeness we should check the `olm.operatorNamespace` annotation as well.